csvcut.py

```diff
@@ -0,0 +1,104 @@
+from argparse import ArgumentParser
+import sys
+
+DESCRIPTION = 'csvcut - selects the specified fields from cvs file'
+EXAMPLES = 'csvcut -f 1,2 stat.txt | less -SR'
+
+def keep_fields(ind, row, output_stream):
+    """
+    Prints the specified fields from .csv file
+    
+    :param indexes: indexes of the required fields represented as a list of indexes
+    :param row: row represented as a list of columns
+    :param output_stream: a stream to print the row with required columns
+    """
+    output_line = ''
+    for i, column in enumerate(row):
+        if i in ind:
+            if i == max(ind):
+                output_line += column + ""
+            elif i < max(ind):
+                
+                output_line += column  + ','
+    
+    output_line += '\n'
+    output_stream.write(output_line)
+    
+
+
+def main():
+    args = parse_args()
+    input_stream = open(args.file, 'r') if args.file else sys.stdin
+    output_stream = open(args.output_file, 'r') if args.output_file else sys.stdout
+
+    columns = input_stream.readline().strip().split(args.separator)
+    first_rows = [columns]
+    
+    
+    
+    fields = args.fields.strip().split(args.separator)
+
+    c = 0
+    if fields[0] != '' and fields[0] != 'all_columns':
+        for i in range(len(fields)):
+            if fields[i].isdigit():
+                c += 1
+    if(c == len(fields)):
+        fields = list(map(int, fields))
+
+    for i in range(1):
+        first_rows.append(input_stream.readline().strip().split(args.separator))
+
+    indexes = []
+    col_counter = 0
+    unique_list = list(set(columns))
+    unique_index = [columns.index(x) for x in unique_list]
+    
+    for i,row in enumerate(first_rows):
+        if i == 0:
+            for j, column in enumerate(row):
+                col_counter += 1
+                if fields[0] == 'all_columns':
+                    indexes.append(j) 
+                elif fields[0] == '':
+                    indexes = []
+                elif isinstance(fields[0], int):
+                    if j in fields:
+                        indexes.append(j)
+                elif isinstance(fields[0], str):
+                    if column in fields:
+                        indexes.append(j)
+
+                          
+        if args.complement:
```

Comment:
Let's calculate all fields related logic once and simplify the code insight this cycle.